src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java

```diff
@@ -107,7 +108,8 @@ public <Request extends ActionRequest, Response extends ActionResponse> void app
       // Source is returned in response hits by default. If disabled by the user, overwrite and enable
       // in order to access document contents for reranking, then suppress at response time.
       boolean suppressSourceOnResponse = false;
-      if (searchRequest.source().fetchSource() != null && !searchRequest.source().fetchSource().fetchSource()) {
+      if (searchRequest.source() != null && searchRequest.source().fetchSource() != null &&
+              !searchRequest.source().fetchSource().fetchSource()) {
```

Comment:
Is it a problem that we have to reach so deep into the SearchRequest?

Reply:
Yeah... this is just how the core SearchRequest class is structured:

* SearchRequest holds (mostly optional) high-level flow-control type stuff (routing param, scroll state, timeout), plus a reference to the SearchSourceBuilder (which never actually builds a SearchSource, but 🤷 ).
* SearchSourceBuilder holds most of what we think of as "the search request" -- the query, highlight options, sort options, from+size, etc. including the FetchSourceContext.
* The FetchSourceContext effectively represents return field options -- `fetchSource` is a boolean where `true` means "Yes, return all fields" and `false` means "No, return no fields". You can also specify includes/excludes (which may include wildcards) to fetch a subset of fields.